### ISSUE TITLE
Feature Request:  Add support for loading exporter list from nbconvert

### ISSUE THREAD

**Issue by benthomasson** [jupyterlab/jupyterlab#5015]
In the JupyterLab UI it appears that the exporter list is hard coded.   Ansible Jupyter Kernel (https://github.com/ansible/ansible-jupyter-kernel) registers new exporters for nbconvert.  Those exporters are not appearing in the JupyterLab UI.  I added those exporters manually here: https://github.com/benthomasson/jupyterlab/commit/b1dfbf580255863032159f860d386cd4cb035d7d and it works fine. I would prefer that the exporter list is dynamically loaded as was done in Jupyter Notebook.

**Comment by blink1073** [comment]
Hi @benthomasson, thanks for the report.  

Dev note, here's how it could be done:
- Add the list of exporters to `page_config` using [this function](https://github.com/jupyter/notebook/blob/64f02d86219789363ef22a6ed380d8541737dc78/notebook/notebook/handlers.py#L18) and populating [here](https://github.com/jupyterlab/jupyterlab_launcher/blob/bf41073898e5096332e122763a0a837cc7faa7d8/jupyterlab_launcher/handlers.py#L73)
- Use `PageConfig.getOption()` [here](https://github.com/jupyterlab/jupyterlab/blob/e33fcb9238230696e782c6d734888a71fac09d02/packages/notebook/src/celltools.ts#L658)

**Comment by jasongrout** [comment]
Feel free to submit a PR, if you'd like! @blink1073, do you think this is a good first issue?

**Comment by blink1073** [comment]
I didn't tag it, because it involves both Python and JS, and spans both repos.

**Comment by ian-r-rose** [comment]
See also #3837.

I think this might not even require a change to the server side. The `api/nbonvert` root endpoint returns a list of export formats. When I check that, I see
```json
{"custom": {"output_mimetype": ""}, "html": {"output_mimetype": "text/html"}, "slides": {"output_mimetype": "text/html"}, "latex": {"output_mimetype": "text/latex"}, "pdf": {"output_mimetype": "text/latex"}, "markdown": {"output_mimetype": "text/markdown"}, "python": {"output_mimetype": "text/x-python"}, "rst": {"output_mimetype": "text/restructuredtext"}, "notebook": {"output_mimetype": "application/json"}, "script": {"output_mimetype": ""}}
```

**Comment by timkpaine** [comment]
yeah I use this in lantern:
https://github.com/timkpaine/lantern/blob/master/lantern/extensions/hideinput/extension.py#L6

https://github.com/timkpaine/lantern/blob/master/lantern/extensions/hideinput/exporters.py#L22

**Comment by dharmaquark** [comment]
I'd be interested in helping with this! seems like it would be pretty straightforward to add nbconvert to services, with a new Nbconvert class getter for the export formats list that we could use in the [createNBConvertSelector](https://github.com/jupyterlab/jupyterlab/blob/e33fcb9238230696e782c6d734888a71fac09d02/packages/notebook/src/celltools.ts#L658) function to replace the hard-coded list

**Comment by dharmaquark** [comment]
so it looks like the export command options, which set the menu options (File >Export As... >), also uses a separate hard-coded [list](https://github.com/jupyterlab/jupyterlab/blob/ec986f4bf6c813b6facbdd551765193e63fa7f12/packages/notebook-extension/src/index.ts#L217). I'm thinking it might make sense to also have this map this to the same nbconvert list of export formats.
Any objections to having both the cell inspector convert options and export command options use the same nbconvert list? (I can treat these changes as two separate PRs, if that makes things easier to review!)

**Comment by ian-r-rose** [comment]
I think that makes sense! Feel free to do two PRs or one, whichever works for you.

**Comment by ian-r-rose** [comment]
Fixed by #5217. Thanks @dharmaquark!
